### PR TITLE
trojan-go 0.10.0 (new formula)

### PR DESCRIPTION
**Pull request by Misaka13514**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

## Questions need help
- ~~Some functions of the software need to be used with other software. It needs a http web server and ssl certificates to test. **How can I write that in `test do`?**~~
  > [Source code on GitHub](https://github.com/p4gefau1t/trojan-go)
  > Here are some [English docs](https://trojan-gfw.github.io/trojan/config) for a similar software.

  (I think: A server without new dependencies and special permission requirements which starts easily should be added. Or add a reliable remote http server.)
  (I don't know whether self-signed ssl certificates can work.)
- ~~I don't know how to set env for the software to read required files. Does it provide this function?~~
- ~~Should I add `def plist`?~~
- ~~Should I add `bottle do ... end`? I'm not very sure whether this is necessary.~~
- ~~Trojan-go uses the pre-release setting on all releases so the audit fails.~~
- Is there any other problem in my code?

## Software reference
- clash
- v2ray
- xray

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for your PR, @Misaka13514.

For future reference, please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings. Refer to the [documentation](https://docs.brew.sh/Formula-Cookbook#commit) or the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md) for more details.

When you make further changes to your PR, please squash them all into one commit and modify the commit message to read
```
trojan-go 0.8.3 (new formula)
```

**Comment by Misaka13514** [comment]
Because the documentation of this software is not in English, if there are parts that you can't understand, you can show me the original Chinese text and I will try my best to explain it.

**Comment by Loyalsoldier** [comment]
Trojan-Go v0.9.0 has been release. BTW, it uses `geoip.dat` & `geosite.dat` as v2ray does, I suggest you take v2ray's formula as reference.

**Comment by Misaka13514** [comment]
Thank you @carlocab . I used to write code in `home-brew-cask` in the past, I will pay attention to differences between repositories.

**Comment by Misaka13514** [comment]
Thank you @Loyalsoldier . I added `geoip.dat` & `geosite.dat`, but I don't know how to set env for trojan-go to read them.

Any idea for my questions?

**Comment by Loyalsoldier** [comment]
> Thank you @Loyalsoldier . I add `geoip.dat` & `geosite.dat`, but I don't know how to set env for trojan-go to read them.
> 
> Any idea for my questions?

Just set an environment variable called `TROJAN_GO_LOCATION_ASSET` like this one in V2Ray formula:

https://github.com/Homebrew/homebrew-core/blob/2f0d590c54baf3a25cfd06351eccb15b9742fb0e/Formula/v2ray.rb#L44-L45

BTW, Trojan-Go v0.10.0 has been released.

**Comment by Loyalsoldier** [comment]
BTW, I think example config file is needed for Trojan-Go.

**Comment by Misaka13514** [comment]
> BTW, I think example config file is needed for Trojan-Go.

In my point of view, it's not necessary to add example config, like `xray` and `clash`.

If I add the example configs.(like this)
```ruby
    pkgetc.install "example/client.json"
    pkgetc.install "example/client.yaml"
    pkgetc.install "example/server.json"
    pkgetc.install "example/server.yaml"
```

Users don't know where are the config files to edit. Besides, some warning will pop up during the uninstall process.
```text
Warning: The following trojan-go configuration files have not been removed!
If desired, remove them manually with `rm -rf`:
  /usr/local/etc/trojan-go
  /usr/local/etc/trojan-go/client.json
  /usr/local/etc/trojan-go/client.yaml
  /usr/local/etc/trojan-go/server.json
  /usr/local/etc/trojan-go/server.yaml
```

**Comment by Misaka13514** [comment]
So... How to write the `test` code? Please help me. 🤔

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:17]: This is fine if you're leaving it like this for testing, but these will need a versioned URL and checksums before merging.

**Review by carlocab — CHANGES_REQUESTED** [review]
I'm afraid I don't know how to help you with your test, but I do have one comment regarding the `resource`s.

**Comment by Misaka13514** [comment]
Can I use `sha256 :no_check` like that in `homebrew-cask`? Only used to download the latest `geoip.dat` & `geosite.dat`, because they can't be `livecheck`ed and they are rarely updated to the latest version, such as them in `v2ray` and `xray`. @carlocab

**Comment by Misaka13514** [comment]
It seems that it cannot be merged in a short time, maybe it will be easier to create in `homebrew-cask` XD

**Comment by carlocab** [comment]
`sha256 :no_check` is unique to Homebrew/cask, and isn't accepted in Homebrew/core. Resources aren't `livecheck`ed, so I wouldn't worry about them becoming out-of-date. What's important is that we're about to verify download integrity.

**Comment by Misaka13514** [comment]
> `sha256 :no_check` is unique to Homebrew/cask, and isn't accepted in Homebrew/core. Resources aren't `livecheck`ed, so I wouldn't worry about them becoming out-of-date. What's important is that we're about to verify download integrity.

Is there a way to update these resources files in time? Through livecheck or other methods
These resources are still out of date in some formula. Few people will open the source code to check their updates.

**Comment by Loyalsoldier** [comment]
> > > Thank you @Loyalsoldier . I add `geoip.dat` & `geosite.dat`, but I don't know how to set env for trojan-go to read them.
> > > Any idea for my questions?
> > 
> > 
> > Just set an environment variable called `TROJAN_GO_LOCATION_ASSET` like this one in V2Ray formula:
> > https://github.com/Homebrew/homebrew-core/blob/2f0d590c54baf3a25cfd06351eccb15b9742fb0e/Formula/v2ray.rb#L44-L45
> > 
> > BTW, Trojan-Go v0.10.0 has been released.
> 
> Hmmm...
> 
> ```ruby
>     (bin/"trojan-go").write_env_script execpath,
>       TROJAN_GO_LOCATION_ASSET: "${TROJAN_GO_LOCATION_ASSET:-#{pkgshare}}"
> ```
> 
> ```
> ==> go build -tags=full
> Error: An exception occurred within a child process:
>   NameError: undefined local variable or method `execpath' for #<Formulary::FormulaNamespace8a7b8a7a4cc63807f78536f0799af33e::TrojanGo:0x00007fd41c2012d0>
> ```

You should search `execpath` in v2ray formula, and it shows up in two lines:

https://github.com/Homebrew/homebrew-core/blob/2f0d590c54baf3a25cfd06351eccb15b9742fb0e/Formula/v2ray.rb#L33-L45

**Comment by Loyalsoldier** [comment]
> > Is there a way to update these resources files in time?
> 
> Make them into separate formulae and make sure those are updated.

Is it acceptable to add non-executable binary into Homebrew-core?

File: Formula/trojan-go.rb
Comment (gromgit) [reply Formula/trojan-go.rb:17]: I suggest something like:
```
  resource "geoip" do
    url "https://github.com/v2fly/geoip/releases/download/202105070029/geoip.dat"
    sha256 "f3aeb80c21fd403c6030988477d0cac7574322b786733d0ecbc24ff4ab55ce93"
  end

  resource "geosite" do
    url "https://github.com/v2fly/domain-list-community/releases/download/20210507022421/dlc.dat"
    sha256 "c08b57f45ad248e3812ff6c453c788c85916cd443d4c11e26ccd1bb6919c928d"
  end
```
Those are the latest as of this writing.

Comment (Misaka13514) [reply Formula/trojan-go.rb:17]: Thank you @gromgit . I'll change these code.
In the long run, these does not seem to be the best solution.

Comment (gromgit) [reply Formula/trojan-go.rb:17]: Then perhaps you should file an issue with upstream. As far as I can tell, these two files are only pulled at install time, with no script or other instructions for updating them over time.

**Comment by Misaka13514** [comment]
Maybe I’m not smart enough, is my code written like this correct? @Loyalsoldier 
```ruby
    execpath = libexec/name
    system "go", "build", *std_go_args, "-tags=full"
    (bin/"trojan-go").write_env_script execpath,
      TROJAN_GO_LOCATION_ASSET: "${TROJAN_GO_LOCATION_ASSET:-#{pkgshare}}"
```

```text
$ trojan-go
/usr/local/bin/trojan-go: line 2: /usr/local/Cellar/trojan-go/0.10.0/libexec/trojan-go: No such file or directory
/usr/local/bin/trojan-go: line 2: exec: /usr/local/Cellar/trojan-go/0.10.0/libexec/trojan-go: cannot execute: No such file or directory
```

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Thanks. I am not a developer of upstream code, this may only be done by them. 🤣

**Comment by gromgit** [comment]
> ```
>     execpath = libexec/name
>     system "go", "build", *std_go_args, "-tags=full"
> ```
If you're trying to compile the final binary to `execpath`, you're missing a flag:
```
    execpath = libexec/name
    system "go", "build", *std_go_args, "-tags=full",
      "-o", execpath
```
Without `-o`, you're building to `bin/"trojan-go"`, and then immediately overwriting it with a script that calls a nonexistent binary.

**Comment by Misaka13514** [comment]
I left it out, thank you for your guidance. @gromgit

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:10]: Remove these two lines.

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:104]: I think the better way to test it is to run `trojan-go -version`, and match the output.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:10]: OK

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:5]: Change `url` and `sha256` to this convention: https://github.com/Homebrew/homebrew-core/blob/1220c789207176204e85bb597fe56530da4f503c/Formula/glooctl.rb#L4-L6

Because building Trojan-Go needs the `version` and `commit` variable passed to go build and test.

File: Formula/trojan-go.rb
Comment (gromgit) [reply Formula/trojan-go.rb:104]: Not a maintainer, but submitting only a version test is highly likely to get your formula rejected: https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:26]:
```suggestion
    ldflags = -X github.com/p4gefau1t/trojan-go/constant.Version=#{tag} -X github.com/p4gefau1t/trojan-go/constant.Commit=#{revision}
    system "go", "build", *std_go_args, "-o", execpath, "-tags=full", "-ldflags", ldflags
```

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:104]: `test` is used to test the function of the software, but it cannot test whether the software is working normally.
If there is any error in the build, it will not be able to test whether the software is working properly.
For example, if you remove the `-tags=full`:
`$ trojan-go -version` shows the same output as normal one.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:5]: Roger that. Thanks.

**Comment by Misaka13514** [comment]
Modified the code according to @Loyalsoldier 's requirements, now there are new problems.

1.
```
brew install --build-from-source trojan-go                         16:29:40
==> Downloading https://github.com/v2fly/geoip/releases/download/202105070029/ge
==> Downloading https://github.com/v2fly/domain-list-community/releases/download
==> Cloning https://github.com/solo-io/gloo.git
==> Checking out tag v1.7.5
HEAD is now at 2bf4a4abd Backport logger fix (#4712)
HEAD is now at 2bf4a4abd Backport logger fix (#4712)
Error: An exception occurred within a child process:
  NameError: undefined local variable or method `tag' for #<Formulary::FormulaNamespace8a7b8a7a4cc63807f78536f0799af33e::TrojanGo:0x00007fc0649d2ca0>
Did you mean?  tap
               @tap
```

2. how to install "example/client.json" now?

**Comment by Loyalsoldier** [comment]
> Modified the code according to @Loyalsoldier 's requirements, now there are new problems.
> 
> 
> ```
> brew install --build-from-source trojan-go                         16:29:40
> ==> Downloading https://github.com/v2fly/geoip/releases/download/202105070029/ge
> ==> Downloading https://github.com/v2fly/domain-list-community/releases/download
> ==> Cloning https://github.com/solo-io/gloo.git
> ==> Checking out tag v1.7.5
> HEAD is now at 2bf4a4abd Backport logger fix (#4712)
> HEAD is now at 2bf4a4abd Backport logger fix (#4712)
> Error: An exception occurred within a child process:
>   NameError: undefined local variable or method `tag' for #<Formulary::FormulaNamespace8a7b8a7a4cc63807f78536f0799af33e::TrojanGo:0x00007fc0649d2ca0>
> Did you mean?  tap
>                @tap
> ```
> 
> 1. how to install "example/client.json" now?

😂 You are cute. You don't need to change the repo. Just follow that tag and revision style.

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:6]:
```suggestion
  url "github.com/p4gefau1t/trojan-go.git",
      tag: "v0.10.0",
      revision: "a9ce6d4ada9143d58e8d1373d23f9ebfe871119e"
```

**Comment by Misaka13514** [comment]
🤣 Get what you mean

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [inline Formula/trojan-go.rb:6]:
```suggestion
      revision: "a9ce6d4ada9143d58e8d1373d23f9ebfe871119e"
```

Comment (Misaka13514) [reply Formula/trojan-go.rb:6]:
```
$ brew install --build-from-source trojan-go
Error: v0.10.0 tag should be a9ce6d4ada9143d58e8d1373d23f9ebfe871119e
but is actually 1d9c46760efe143aa1f9bd7e2f7cf460cb364259
```

Comment (Loyalsoldier) [reply Formula/trojan-go.rb:6]: No need to guess

image

Comment (Misaka13514) [reply Formula/trojan-go.rb:6]: I understand that, I will try cleaning my cache.

**Comment by Misaka13514** [comment]
We are getting closer and closer to success, thank you contributors!

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]:
```
$ trojan-go -version
Trojan-Go v0.10.0
Go Version: go1.16.3
OS/Arch: darwin/amd64
Git Commit: a9ce6d4ada9143d58e8d1373d23f9ebfe871119e

Developed by PageFault (p4gefau1t)
Licensed under GNU General Public License version 3
GitHub Repository:	https://github.com/p4gefau1t/trojan-go
Trojan-Go Documents:	https://p4gefau1t.github.io/trojan-go/
```
well done

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: After trying I still don't know how to separate these two resources into formula and let the program read them. 😞

**Comment by Misaka13514** [comment]
> Is it acceptable to add non-executable binary into Homebrew-core?
Homebrew/discussions/discussions/1424

> Make them into separate formulae and make sure those are updated.
But I don't know how to change. Temporarily I will keep resources in my code.

File: Formula/trojan-go.rb
Comment (Loyalsoldier) [reply Formula/trojan-go.rb]: It's okay to leave them as resources for now.

**Comment by Loyalsoldier** [comment]
I think it's ready for review.

**Comment by Misaka13514** [comment]
What about the `test`? @Loyalsoldier

**Comment by Loyalsoldier** [comment]
> What about the `test`? @Loyalsoldier

I think the version test is okay. It's the upstream's responsibility to keep Trojan-Go to be functional. The downstream(eg, Homebrew) just need to be sure that the Trojan-Go can be built without error.

**Comment by Misaka13514** [comment]
Errors may occur during downstream building, which is the purpose of `test`.

BTW, I think adding an option to force the certificate and http server to be ignored in the upstream will be fine. (?)

I am looking forward to your stable release. @Loyalsoldier 💪

**Comment by Loyalsoldier** [comment]
Stable release is out.

**Comment by Misaka13514** [comment]
Maybe you can add a function to check the configuration file, perhaps it can be a test.

**Comment by Loyalsoldier** [comment]
> Maybe you can add a function to check the configuration file, perhaps it can be a test.

I'm afraid this will not happen.

**Comment by Misaka13514** [comment]
> BTW, I think adding an option to force the certificate and http server to be ignored in the upstream will be fine. (?)

What's your opinion about this

**Comment by Loyalsoldier** [comment]
> > BTW, I think adding an option to force the certificate and http server to be ignored in the upstream will be fine. (?)
> 
> What's your opinion about this

Won't do that for a test.

**Comment by Misaka13514** [comment]
```ruby
  resource "geoip" do
    geoip_hash_url = "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat.sha256sum"
    url "https://github.com/v2fly/geoip/releases/latest/download/geoip.dat"
    sha256 Utils.safe_popen_read("curl", "-L", geoip_hash_url)[0, 64]
  end

  resource "geosite" do
    geosite_hash_url = "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat.sha256sum"
    url "https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat"
    sha256 Utils.safe_popen_read("curl", "-L", geosite_hash_url)[0, 64]
  end
```
Is it acceptable to write like this? 🤣 
@SMillerDev

**Comment by Misaka13514** [comment]
Since it is currently impossible to write `test` code that can test software functions, according to the requirements of upstream developers, I will submit a `test` code containing only version testing.

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:15]: The `url` needs to be versioned and the `sha256` recorded in the formula. We don't want the build to change depending on the day it was built.

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:21]: The `url` needs to be versioned and the `sha256` recorded in the formula. We don't want the build to change depending on the day it was built.

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:26]: Is this different from the version of the formula? If it isn't, you can just drop this line entirely and do
```
-X github.com/p4gefau1t/trojan-go/constant.Version=#{version}
```

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:27]:
```suggestion
    execpath = libexec/name
```

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:33]:
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags.join(" ")), "-o", execpath, "-tags=full"
```

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:34]:
```suggestion
    (bin/"trojan-go").write_env_script execpath,
```

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:71]: Can we use the `service` DSL here? Example:

https://github.com/Homebrew/homebrew-core/blob/b4e38c605e14bef7c2298e092deb8bdc1813394d/Formula/php.rb#L323-L330

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:75]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:6]: Can we use a tarball instead of a Git checkout?

**Comment by Misaka13514** [comment]
You are really experienced @carlocab , I'll change them.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Resolved.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Resolved.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Resolved.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Resolved.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: I modified it. Now there is only a small difference from the previous:

previous
```text
==> Caveats
To have launchd start trojan-go now and restart at login:
  brew services start trojan-go
Or, if you don't want/need a background service you can just run:
  trojan-go -config /usr/local/etc/trojan-go/config.json
```

now
```text
==> Caveats
To start trojan-go:
  brew services start trojan-go
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/trojan-go/bin/trojan-go -config /usr/local/etc/trojan-go/config.json
```

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:6]: `trojan-go` need commit id to show in its version.
```text
$ trojan-go -version
Trojan-Go v0.10.0
Go Version: go1.16.3
OS/Arch: darwin/amd64
Git Commit: a9ce6d4ada9143d58e8d1373d23f9ebfe871119e
```

hmm.. I will try to do that.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:55]: I am afraid I do not know how to write it.
This can hardly be done without requiring upstream code modification.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb]: Resolved.

**Comment by Misaka13514** [comment]
Is there a way to find the revision id `a9ce6d4` by version `v0.10.0`?

**Comment by Misaka13514** [comment]
> > What about the `test`? @Loyalsoldier
> 
> I think the version test is good enough. It's the upstream's responsibility to keep Trojan-Go to be functional. The downstream(eg, Homebrew) just need to be sure that the Trojan-Go can be built without error.

> Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.
> 
> > We want tests that don’t require any user input and test the basic functionality of the application. For example foo build-foo input.foo is a good test and (despite their widespread use) foo --version and foo --help are bad tests.
> 
> Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

This is still a problem we need to solve.

**Comment by Misaka13514** [comment]
> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.

Maybe we can write `test` code in this direction.

File: Formula/trojan-go.rb
Comment (carlocab) [reply Formula/trojan-go.rb:6]: I see. A Git checkout is fine then.

File: Formula/trojan-go.rb
Comment (Misaka13514) [reply Formula/trojan-go.rb:55]: It fails because there is no http server and the cert is invalid.

Comment (carlocab) [reply Formula/trojan-go.rb:55]: Ok, let's try running `trojan-go` without the `fork` + `exec` and check that it returns the expected non-zero exit code and expected error message.

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:28]:
```suggestion
    execpath = libexec/name
    ldflags = %W[
      -X github.com/p4gefau1t/trojan-go/constant.Version=v#{version}
      -X github.com/p4gefau1t/trojan-go/constant.Commit=#{Utils.git_head}
    ]
```
You only use `commit` in one place, so there's no need to define an extra variable for it.

Comment (Misaka13514) [reply Formula/trojan-go.rb:28]: Resolved.

**Comment by Misaka13514** [comment]
I'm sorry to make the code ugly due to the addition of the certificate.

If there is any problem with the code, please point it out.

Thank you all for your help over the past few days.

> I think it's ready for review.

File: Formula/trojan-go.rb
Comment (carlocab) [inline Formula/trojan-go.rb:47]: This relies on https://github.com/Homebrew/brew/pull/11355, so we'll need for the commit from there to land in a tagged release.

Comment (SMillerDev) [reply Formula/trojan-go.rb:47]: `plist_options manual:` won't be needed after that either.

Comment (Misaka13514) [reply Formula/trojan-go.rb:51]: Okay, thanks for reminding.

**Comment by carlocab** [comment]
LGTM, unless @SMillerDev has further feedback. We just need a new `brew` release tag, which should happen today.

**Review by carlocab — APPROVED** [review]
Thanks for your work here, @Misaka13514.

**Pull request closed without merging**
